proposals/dao/balancer_gauge_fix.ts

```diff
@@ -0,0 +1,116 @@
+import hre, { ethers, artifacts } from 'hardhat';
+import { expect } from 'chai';
+import {
+  DeployUpgradeFunc,
+  NamedAddresses,
+  SetupUpgradeFunc,
+  TeardownUpgradeFunc,
+  ValidateUpgradeFunc
+} from '@custom-types/types';
+import { getImpersonatedSigner, overwriteChainlinkAggregator, time } from '@test/helpers';
+import { forceEth } from '@test/integration/setup/utils';
+
+const e18 = '000000000000000000';
+const fipNumber = 'balancer_gauge_fix';
+let pcvStatsBefore;
+
+// Do any deployments
+// This should exclusively include new contract deployments
+const deploy: DeployUpgradeFunc = async (deployAddress: string, addresses: NamedAddresses, logging: boolean) => {
+  // Deploy balancer gauge staker implementation
+  const balancerGaugeStakerFactory = await ethers.getContractFactory('BalancerGaugeStaker');
+  const balancerGaugeStakerImplementation = await balancerGaugeStakerFactory.deploy(
+    addresses.core // address _core
+  );
+  await balancerGaugeStakerImplementation.deployTransaction.wait();
+  logging && console.log('balancerGaugeStakerImplementation: ', balancerGaugeStakerImplementation.address);
+
+  // Deploy balancer gauge staker proxy
+  const ProxyFactory = await ethers.getContractFactory('TransparentUpgradeableProxy');
```

Comment:
We're using a proxy pattern to future proof any future need to upgrade?

Reply:
Balancer will hardcode the boost delegation from our veBAL locking contract to this address (possibly, if it turns out to be a workable solution on their side). So we want a proxy to avoid asking them to "change the address to which the delegation is hardcoded"

Reply:
Makes sense, like it